RecoDataProducts/inc/TrkCaloHitSeed.hh

```diff
@@ -13,37 +15,70 @@
 #include <Rtypes.h>
 namespace mu2e {
   struct TrkCaloHitSeed {
-    TrkCaloHitSeed() :  _trklen(0), _hitlen(0), _cdoca(0),_rerr(0)  {}
-    // construct from the information
+    TrkCaloHitSeed() :  _cdepth(0), _cdoca(0), _ctoca(0), _ctocavar(0), _dt(0), _tresid(0), _tresidmvar(0), _tresidpvar(0), _trklen(0), _hitlen(0),  _time(0), _terr(0), _rerr(0) {}
```

Comment:
Are you suggesting a constructor that takes all the data members as arguments?  That is error prone when there are lots of struct members, as its hard to see if the order of arguments is wrong.  Filling the arguments later requires each data member be filled by name, which I find clearer.    It's true that leaves open the possibility of omitting setting a variable; the default constructor settings make that obvious (downstream).

Reply:
No.  If a data member is intialized in the declaration, like:

  int var = 2;

Then it can be omitted from the intializer list and they will intialized to the correct value in the correct order.

This definitely works for integral types and I think it works for floating types to; I will check for float.  I believe that the rhs obeys the same rules as the rhs of a constexpr.

Reply:
I just checked - it does work for floating types too.

Reply:
Doing it this way also ensures that, when there are multiple c'tors, this produces a single point of maintenance for data members that should be initialized to the same value in all c'tors; of one c'tor of several wants to initialize to a different value they can override it in the intializer list.

Reply:
OK I see now.  It's a valid suggestion, I'll give it a try.